perf_test/batched/KokkosBatched_Test_Gemm_Cuda.cpp

```diff
@@ -365,9 +365,10 @@ namespace KokkosBatched {
 
             typedef Functor<view_type,AlgoTagType,VectorLength> functor_type;
             typedef Kokkos::Impl::ParallelFor<functor_type,policy_type,DeviceSpaceType> parallel_for_type;
-              
+            typedef Kokkos::LaunchBounds<> launch_bounds_type;
+
             const int team_size = 
-              Kokkos::Impl::cuda_get_opt_block_size<parallel_for_type>(functor_type(), VectorLength, 0, 0)/VectorLength;
+              Kokkos::Impl::cuda_get_opt_block_size<parallel_for_type,launch_bounds_type>(functor_type(), VectorLength, 0, 0)/VectorLength;
```

Comment:
@kyungjoo-kim changes look fine but I think Kokkos already sets the launch bounds template parameter by default to `Kokkos::LaunchBounds<>` so this shouldn't be necessary unless you want to explicitly control this. Did you run into compilation errors without using this?

Reply:
No. kokko does not put a default launch bounds and that is why I make these changes. If kokkos has a plan to do so, this PR should be modified. Anyway at this moment, kokkos does not have a default template parameter.

Reply:
at least what I see from the kokkos develop branch. 

https://github.com/kokkos/kokkos/blob/f7065cacbec39923026561407b35cceabf0d0720/core/src/Cuda/Kokkos_Cuda_Internal.hpp#L58

as this under impl space, I  just follow the changes made here.

Reply:
I don't get it. Yeah we are not putting LaunchBounds<> in by default, but omitting them has the same effect as adding LaunchBounds<>.

Reply:
This is the error occurred while compiling and it requires launchboudns.
```
error: no instance of function template "Kokkos::Impl::cuda_get_max_block_size" matches the argument list 
```

Reply:
See below: you shouldn't really use the implementation functions, you should go through the public interface.

Reply:
I didn't realize the `cuda_get_opt_block_size` routine should deduce the `LaunchBounds` template parameter from the return type `CudaGetOptBlockSize<...>::get_block_size(...)` (which is partial specialized on LaunchBounds) - so if a LauchBounds template parameter is not provided to the `cuda_get_opt_block_size` then it should hit either the partial specialization of line 205 or 235, where the partial specialization is for `LaunchBounds<>`.